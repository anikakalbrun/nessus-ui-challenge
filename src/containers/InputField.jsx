import React, {Component} from 'react';
import {connect} from "react-redux";

import {setInputValue} from '../actions';
import {fetchServerConfigurations} from '../actions'

export class InputField extends Component {

    componentDidUpdate(prevProps) {
        if(prevProps.inputValue !== this.props.inputValue) {
            this.props.fetchServerConfigurations('api.tenable.io', this.props.inputValue)
        }
    }

    handleChange = (event) => {
        if(event.target.value.length > 4) return;
        this.props.setInputValue(event.target.value);
    };

    render() {
        const {inputValue} = this.props;
        return (
            <div className="form-group InputField">
                <label className='col-lg-12' style={{padding: 0}}>Enter number of expected records:</label>
                <div className='col-lg-3' style={{padding: 0}}>
                <input
                    name="host"
                    className="form-control"
                    type="number"
                    min={0}
                    max={9999}
                    step={1}
                    onChange={this.handleChange}
                    value={inputValue}
                />
                <small className="form-text text-muted">Table data will be generated</small>
                </div>
            </div>
        );
    }
}

export default connect(state => ({inputValue: state.inputValue}), {setInputValue, fetchServerConfigurations})(InputField);