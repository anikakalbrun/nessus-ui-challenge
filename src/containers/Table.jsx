import React, {Component} from 'react';
import {array} from 'prop-types';
import {connect} from "react-redux";
import {BootstrapTable, TableHeaderColumn} from 'react-bootstrap-table';
import 'react-bootstrap-table/dist/react-bootstrap-table-all.min.css';

import {fetchServerConfigurations} from '../actions';

const options = {
    sizePerPageList: [{
        text: '5', value: 5
    }, {
        text: '10', value: 10
    }, {
        text: '25', value: 25
    }]
};
const filterType = {type: 'TextFilter', delay: 1000};

export class Table extends Component {

    static propTypes = {
        data: array
    };

    static defaultProps = {
        data: []
    };

    componentWillMount() {
        this.props.fetchServerConfigurations('api.tenable.io', 2)
    }

    render() {
        const {data} = this.props;
        return (
            <BootstrapTable data={data}
                            pagination={true}
                            options={options}
                            hover={true}
                            exportCSV
                            csvFileName={'table-export.csv'}>
                <TableHeaderColumn dataField='name'
                                   filter={filterType}
                                   isKey>Server</TableHeaderColumn>
                <TableHeaderColumn dataField='hostname'
                                   filter={filterType}>Hostname</TableHeaderColumn>
                <TableHeaderColumn dataField='port'
                                   filter={filterType}>Port</TableHeaderColumn>
                <TableHeaderColumn dataField='username'
                                   filter={filterType}>Username</TableHeaderColumn>
            </BootstrapTable>
        );
    }
}

function mapStateToProps(state) {
    return {
        data: state.data.configurations
    };
}

export default connect(mapStateToProps, {fetchServerConfigurations})(Table);