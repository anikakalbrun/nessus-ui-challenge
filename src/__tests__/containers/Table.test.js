import React from 'react';
import { Table } from '../../containers/Table';

let wrapper;
const fetchServerConfigurations = jest.fn();
const props = {
    data: [{
        "name": "host1",
        "hostname": "nessus-abc-lab.com",
        "port": "8888",
        "username": "admin"
    }],
    value : "",
    fetchServerConfigurations
};

describe('Table', () => {

    beforeEach(() => {
        wrapper = shallow(<Table {...props}/>);
    });

    it('renders the BootstrapTable component', () => {
        expect(wrapper.find('BootstrapTable').length).toEqual(1);
    });

    it('renders 4 columns', () => {
        expect(wrapper.find('TableHeaderColumn').length).toEqual(4);
    });

    it('first column proper title and filter', () => {
        expect(wrapper.find('TableHeaderColumn').at(0).html()).toMatchSnapshot();
    });

    it('second column proper title and filter', () => {
        expect(wrapper.find('TableHeaderColumn').at(1).html()).toMatchSnapshot();
    });

    it('third column proper title and filter', () => {
        expect(wrapper.find('TableHeaderColumn').at(2).html()).toMatchSnapshot();
    });

    it('fourth column proper title and filter', () => {
        expect(wrapper.find('TableHeaderColumn').at(3).html()).toMatchSnapshot();
    });

    it('pagination is set to true', () => {
        expect(wrapper.find('BootstrapTable').first().props().pagination).toEqual(true);
    });

    it('hover is set to true', () => {
        expect(wrapper.find('BootstrapTable').first().props().hover).toEqual(true);
    });

    it('exportCSV is set to true', () => {
        expect(wrapper.find('BootstrapTable').first().props().exportCSV).toEqual(true);
    });

    it('has proper csvFileName ', () => {
        expect(wrapper.find('BootstrapTable').first().props().csvFileName).toEqual('table-export.csv');
    });



});
