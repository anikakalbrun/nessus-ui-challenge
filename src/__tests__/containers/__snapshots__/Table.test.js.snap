// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table first column proper title and filter 1`] = `"<th class=\\"\\" style=\\"text-align:left\\" title=\\"Server\\">Server<div><input type=\\"text\\" class=\\"filter text-filter form-control\\" placeholder=\\"Enter Server...\\" value=\\"\\"/></div></th>"`;

exports[`Table fourth column proper title and filter 1`] = `"<th class=\\"\\" style=\\"text-align:left\\" title=\\"Username\\">Username<div><input type=\\"text\\" class=\\"filter text-filter form-control\\" placeholder=\\"Enter Username...\\" value=\\"\\"/></div></th>"`;

exports[`Table second column proper title and filter 1`] = `"<th class=\\"\\" style=\\"text-align:left\\" title=\\"Hostname\\">Hostname<div><input type=\\"text\\" class=\\"filter text-filter form-control\\" placeholder=\\"Enter Hostname...\\" value=\\"\\"/></div></th>"`;

exports[`Table third column proper title and filter 1`] = `"<th class=\\"\\" style=\\"text-align:left\\" title=\\"Port\\">Port<div><input type=\\"text\\" class=\\"filter text-filter form-control\\" placeholder=\\"Enter Port...\\" value=\\"\\"/></div></th>"`;
